libosmscout/include/osmscout/util/GeoBox.h

```diff
@@ -110,6 +110,25 @@ namespace osmscout {
     }
 
     /**
+     * Create new GeoBox from intersection of this with other
+     * If not Intersects, invalid GeoBox is returned
+     * @param other
+     */
+    GeoBox Intersection(const GeoBox& other) const
```

Comment:
Usually, all what's defined in the class headers is inline. But, since you already define all with "inline", I'll add that as well. And the max-include, sorry for missing it.